pkg/installer/custominstallconfig.go

```diff
@@ -54,6 +53,24 @@ func (m *manager) applyInstallConfigCustomisations(ctx context.Context, installC
 		dnsConfig.GatewayDomains = append(m.env.GatewayDomains(), m.oc.Properties.ImageRegistryStorageAccountName+".blob."+m.env.Environment().StorageEndpointSuffix)
 	}
 
+	// _userDefinedRoutingEnabled will return a boolean determining whether
+	// the AFEC feature Microsoft.RedHatOpenShift/UserDefinedRouting is enabled
+	_userDefinedRoutingEnabled := func(m *manager) bool {
+		for _, feature := range m.sub.Properties.RegisteredFeatures {
+			if feature.Name == api.FeatureFlagUserDefinedRouting && feature.State == string(api.SubscriptionStateRegistered) {
+				return true
+			}
+		}
+		return false
+	}
+	// Determine whether this cluster is a noPublicIP cluster
+	// At creation we can assume a single ingress profile TODO: Maybe `IngressProfiles` should be a map?
+	m.oc.Properties.NetworkProfile.UserDefinedRouting = api.UserDefinedRoutingDisabled
+	if m.oc.Properties.APIServerProfile.Visibility == api.VisibilityPrivate &&
+		m.oc.Properties.IngressProfiles[0].Visibility == api.VisibilityPrivate && _userDefinedRoutingEnabled(m) {
+		m.oc.Properties.NetworkProfile.UserDefinedRouting = api.UserDefinedRoutingEnabled
+	}
```

Comment:
This shouldn't go here.  This function is specifically for creating custom installation configuration, and we're not doing that with the UserDefinedRouting things.  We should be doing this elsewhere.

Reply:
Make it it's own step within `pkg/cluster/install.go`.

Call it something like: `populateUserDefinedRouting` and utilize the existing `pkg/util/feature/subscription.go` function called `IsRegisteredForFeature` instead of writing your own.

Reply:
+1